src/Tools/CodeGenerator/Featurizers.yaml

```diff
@@ -1,4 +1,95 @@
 featurizers:
+  # ----------------------------------------------------------------------
+  - name: AnalyticalRollingWindowFeaturizer
+    estimator_name: AnalyticalRollingWindowEstimator
+    release_version: 0.4.0
+    num_output_columns: 1
+    has_dynamic_output: false
+    description: |-
+        Calculates data based on a rolling window. Currently supports mean. Works for general time series data sets.
+        
+        Input type for this featurizer is a tuple of time point of current row, the grain columns, and target value column.
+
+        C++-style pseudo signature:
+        template <typename T> std::vector<double> execute(std::tuple<std::chrono::system_clock::time_point, std::vector<std::string>, T> const &value);
+
+        Examples:
+            A simple example would be maxHorizon = 1, window size = 2, and we want to take the mean.
+
+            +----------------+-----------+-------+-------------------+
+            |time(YYYY-MM-DD)| grain     | target| target_mean       |
+            +================+===========+=======+===================+
+            | 2017-01-01     | A         | 10    | [NAN]             |
+            +----------------+-----------+-------+-------------------+
+            | 2017-02-01     | A         | 4     | [10]              |
+            +----------------+-----------+-------+-------------------+
+            | 2017-03-01     | A         | 6     | [7]               |
+            +----------------+-----------+-------+-------------------+
+            | 2017-04-01     | A         | 11    | [5]               |
+            +----------------+-----------+-------+-------------------+
+
+            A more complex example would be, assuming we have maxHorizon = 2, window size = 2, min window periods = 2, and we want the mean
+            +----------------+-----------+-------+-------------------+
+            |time(YYYY-MM-DD)| grain     | target| target_max        |
+            +================+===========+=======+===================+
+            | 2017-01-01     | A         | 10    | [NAN, NAN]        |
+            +----------------+-----------+-------+-------------------+
+            | 2017-02-01     | A         | 4     | [NAN, NAN]        |
+            +----------------+-----------+-------+-------------------+
+            | 2017-03-01     | A         | 6     | [NAN, 7]          |
+            +----------------+-----------+-------+-------------------+
+            | 2017-04-01     | A         | 11    | [7, 5]            |
+            +----------------+-----------+-------+-------------------+

```

Comment:
I think another example can be added here with window size is not the same as min window periods.

Reply:
The first example is that way, window size is 2 but it starts doing the math as soon as 1 row is available. Though looking again at my yaml I have the minWindowCount set to required. It should be optional and defaults to just 1. I will update it with that change.